README.md

```diff
@@ -1,70 +1,298 @@
+   </td>
+  </tr>
+  <tr>
+   <td>3D Secure
+   </td>
+   <td>You must enable 3D Secure for secure card transactions. Please contact us if you wish to disable this feature in the Production environment.
+   </td>
+  </tr>
+  <tr>
+   <td>Save Card
+   </td>
+   <td>This will allow your customer to save their card on the payment popup, for faster payment flow on the following purchase.
+   </td>
+  </tr>
+</table>
+
+
+
+## Magento Midtrans Notifications Handling
+
+Login to your [Midtrans Account](https://dashboard.midtrans.com/settings/vtweb_configuration), select your environment (sandbox/production), go to menu **settings** > **configuration** and feel out the following URL fields:
+
+
+
+*   Payment Notification URL:
+
+    `https://[your-site-url]/snap/payment/notification`
+
+*   Finish Redirect URL:
+
+    `https://[your-site-url]/snap/index/finish`
+
+*   Unfinish Redirect URL:
+
+    `https://[your-site-url]/snap/index/finish`
+
+*   Error Redirect URL:
+
+    `https://[your-site-url]/snap/index/finish`
+
+
+## How to online refund transaction
+
+You can request refunds either in the [Midtrans Dashboard](https://dashboard.midtrans.com/transactions) or in the Magento admin. After a refund is issued, it cannot be cancelled or undone. Before you trigger this request, make sure that the refund amount and any other details are correct. The online refund function is available only for payment method gopay and credit card.
+
+If you make a refund in the Midtrans Dashboard, a Refund notification is sent to Magento, set transaction state to CLOSE and for now is not created the credit memo.
+
+
+### Request refund from Magento Admin:
+
+
+
+1. Log in to your Magento admin panel. 
+2. In the menu, go to **Sales** > **Orders**. This opens the order overview page. 
+3. Click on the order you want to refund.
+4. In the **Order list View** left-hand navigation sidebar, click **Invoices**tab.
+5. In the invoice list page, selected the order. click the view button on invoice you need to request online refund.
+6. Click **Credit Memo** on the top-right corner of the page. ****
+7. In the **New Memo for Invoice** page, scroll down to the **Refund Totals** section.
+8. In this section, you can request a online **Refund** or a **Refund Offline**.
+    *   **Refund:** This option request refund Online to Midtrans, Midtrans automatically send refund notification and changes to **Closed**.
+    *   **Refund Offline**: An offline refund does not trigger request refund to midtrans it’s only refund in Magento side. You need to take action and carry out the refund manually from Midtrans dashboard.
+
+After a refund operation, the order status changes to **Closed**. This change is controlled by the Magento system, and we cannot influence it. The status change does not mean that the refund was carried out successfully on our side. If the transaction status in Midtrans dashboard changes to **REFUND**, then the refund went through successfully
```

Comment:
> we cannot influence it
who are we? Midtrans? Midtrans Notification?

not `influence` but `change/modify`

> The status change may not mean that the refund has carried out successfully on our side.

> When the transaction status in  ...

Reply:
done